templates/server-config-configmap.yaml

```diff
@@ -14,43 +14,7 @@ metadata:
     app.kubernetes.io/managed-by: {{ .Release.Service }}
 data:
   extraconfig-from-values.hcl: |-
-    disable_mlock = true
-  {{- if or (eq .mode "ha") (eq .mode "standalone")}}
-
-    {{- range $key, $value := (index .Values.server .mode).config }}
-      {{- if kindIs "map" $value }}
-        {{- /* The value contains an stanza. Expand into name and paramenters */ -}}
-
-        {{- $named_stanza := false }}
-        {{- range $i := (values $value ) }}
-          {{- if kindIs "map" $i }}
-            {{- $named_stanza = true }}
-          {{- end }}
-        {{- end }}
-
-        {{- if $named_stanza }}
-          {{- /* Value contains a nested map. This is a named stanza */}}
-          {{- range $name, $parameters := $value }}
-            {{- if kindIs "map" $parameters }}
-    {{ $key }} {{ $name | quote }} {
-              {{- range $parameter_key, $parameter_value := $parameters }}
-      {{ $parameter_key }} = {{ $parameter_value | quote }}
-              {{- end }}
-            {{- end }}
-          {{- end }}
-        {{- else }}
-          {{- /* Unnamed stanza */}}
-    {{ $key }} {
-          {{- range $parameter_key, $parameter_value := $value }}
-      {{ $parameter_key }} = {{ $parameter_value | quote }}
-          {{- end }}
-        {{- end }}
-    }
-      {{- else }}
-    {{ $key }} = {{ $value }}
-      {{- end }}
-    {{- end }}
-  {{ end }}
+{{ .Values.server.standalone.config | toPrettyJson | indent 4 }}
```

Comment:
We'll need to support all modes and their respective configurations here.

Reply:
Yeah, sure. Was just thinking in a very loudly way ;)

Last commit has the actual code, taking the config from standalone or from dev and merging the default parameter.

In the future the way the config and the mode is defined could be improved to have a more straightforward values file (modify a single key to change the mode for example).